Create a service for products

spmse opened the pull request:
Adds a Service and corresponding `vuex` Store for Products. Closes #15 
Please wait before merge until WIP is resolved.

Commit:
delete markdown file

Commit:
Merge branch 'dev' of github.com:ProjektKuro/kuro-web into create-a-service-for-products

Commit:
add product service

Commit:
create crud operations-service for the api. Closes #48

Commit:
create a service for products. Resolves #15

Commit:
create product store

Commit:
add application wide interfaces

Commit:
add vuex, vuex-simple and include store in app

Commit:
add service and store for products

spmse commented:
WIP will be resolved when #51 can be closed.

Commit:
update in store

Commit:
remove comment

spmse commented:
commit 2f6a402 fixes #51

Commit:
Merge branch 'dev' of github.com:ProjektKuro/kuro-web into create-a-service-for-products

tlahmann commented on src/services/api/productService.ts:
```diff
@@ -0,0 +1,22 @@
+export class ProductService {
+    
+    static getAllProducts = async (): Promise<AxiosResponse> => {
+        return await BaseApiService.instance.read("/products");
```
Either embed this line in a try catch block or split this line into two. [ES-lint rule](https://github.com/eslint/eslint/blob/master/docs/rules/no-return-await.md) & [another example](https://jakearchibald.com/2017/await-vs-return-vs-return-await/#return-awaiting).

The same applies to all return await calls.

tlahmann commented on src/services/api/base.ts:
```diff
@@ -0,0 +1,45 @@
+  }
+
+  read = async (resource: string = ""): Promise<AxiosResponse> => {
+    return await this.http.get(resource);
```
Either embed this line in a try catch block or split this line into two. [ES-lint rule](https://github.com/eslint/eslint/blob/master/docs/rules/no-return-await.md) & [another example](https://jakearchibald.com/2017/await-vs-return-vs-return-await/#return-awaiting).

The same applies to all return await calls.

Commit:
remove Map from detailed Product view

Commit:
rearrange things in order to show productdetails straightly

tlahmann commented:
@SvenPatrickMeier, Is this ready to merge?

spmse commented:
@tlahmann Almost ready. Will finish tomorrow!

Commit:
add new layout to products page

also adds more soft hyphen character to keep this working throughout all languages

Commit:
update styles to remove layout breaks

layout had massive breakouts with different sizes of the screen

Commit:
Merge branch 'dev' into create-a-service-for-products

spmse commented:
Ready to merge after review.

Commit:
add margin to content again.
